libs/common/src/errors/invalid-deal-id.error.ts

```diff
@@ -0,0 +1,8 @@
+export class InvalidDealIdError extends Error {
+  constructor(dealId?: string) {
+    const message = dealId ? `Invalid deal ID: ${dealId}` : 'Invalid deal ID';
+    super(message);
```

Comment:
Might've commented twice, but would prefer
```suggestion
import { ApiError } from './api.error';

export class InvalidDealIdError extends ApiError {
  constructor(dealId?: string) {
    const message = dealId ? `Invalid deal ID: ${dealId}` : 'Invalid deal ID';
    super({ message, status: 400 });
```